ghcide/src/Development/IDE/Core/FileStore.hs

```diff
@@ -224,7 +224,7 @@ setFileModified recorder vfs state saved nfp = do
           CheckOnSave -> saved
           _           -> False
     join $ atomically $ recordDirtyKeys (shakeExtras state) GetModificationTime [nfp]
```

Comment:
this needs to be pushed into the `restartShakeSession`?

Reply:
right, I'll need track those usage down.

Reply:
I think we can just delete the `recordDirtyKeys` function and replace it by supplying arguments to `restartShakeSession`

Reply:
done